big-red-shuttle/StopsViewController.swift

```diff
@@ -60,8 +62,43 @@ class StopsViewController: UIViewController, UITableViewDelegate, UITableViewDat
         let locations = stops.map { $0.getCoordinate() }
         setLocations(locations: locations)
         viewIsSetup = true
+        
+        //observe gps data changes
+        system().gps().addObserver(observer: self)
+        Location.shared.fetchedUserLocationCompletionBlock = {
+            if !system().gps().registeredToLogLocation {
+                system().api().registerUserToLogLocation(key: "fedorko", success: { (json: JSON?) in
+                    system().gps().startLoggingShuttleLocation()
+                    system().gps().startFetchingShuttleLocation()
+                }, failure: nil)
+            }
+        }
+    }
+    
+    //MARK: - Shuttle Bus GPS Delegate Protocol Methods
+    func gps(gps: GPS, movedToCoordinate coordinate: Coordinate) {
```

Comment:
more of a convention to put the gps into here as this is part of the GPS delegate protocol (kind of like table views pass the table view into the protocol method).

My idea was to have GPS be a global background process class which fetches/logs gps info in the background and notifies whoever wants to listen